### PR TITLE
Docs: Improve multiprocessing.SharedMemory reference

### PR DESCRIPTION
**Pull request by erlend-aasland**
Align the multiprocessing shared memory docs with Diatáxis's
recommendations for a reference.

- use a parameter list for the `SharedMemory.__init__()` argument spec
- use the imperative mode in reference sections
- use versionadded, not versionchanged for added parameters
- reflow some touched lines according to SemBr

----
📚 Documentation preview 📚: https://cpython-previews--114093.org.readthedocs.build/en/114093/library/multiprocessing.shared_memory.html

### CODE REVIEW

**Comment by erlend-aasland** [comment]
I'll split out the Sphinx corrections in a separate PR.

File: Doc/library/multiprocessing.shared_memory.rst [hunk 1]
```diff
@@ -38,7 +38,8 @@ copying of data.
 
 .. class:: SharedMemory(name=None, create=False, size=0, *, track=True)
 
-   Creates a new shared memory block or attaches to an existing shared
+   Create an instance of the :class:`!SharedMemory` class to either
+   create a new shared memory block or attache to an existing shared
```
Comment (erlend-aasland) [inline Doc/library/multiprocessing.shared_memory.rst]:
```suggestion
   Create an instance of the :class:`!SharedMemory` class for either
   creating a new shared memory block or attaching to an existing shared
```

File: Doc/library/multiprocessing.shared_memory.rst [hunk 2]
```diff
@@ -51,41 +52,48 @@ copying of data.
...
+      shared memory segments is handled automatically among these processes.
+      Python processes created in any other way will receive their own
+      resource tracker when accessing shared memory with *track* enabled.
+      This will cause the shared memory to be deleted by the resource tracker
+      of the first process that terminates.
+      To avoid this issue, users of :mod:`subprocess` or standalone Python
+      processes should set *track* to ``False`` when there is already another
+      process in place that does the bookkeeping.
+      *track* is ignored on Windows, which has its own tracking and
+      automatically deletes shared memory when all handles to it have been closed.
```
Comment (erlend-aasland) [inline Doc/library/multiprocessing.shared_memory.rst:89]: This is a _very_ long description for a parameter. We should extract some of this information into a note or something else. Well, for another PR.

Comment (erlend-aasland) [reply Doc/library/multiprocessing.shared_memory.rst:89]: Perhaps create a small "explanation" section for this.

**Review by hugovk — APPROVED** [review]
Approving with the suggested change. Thanks!

**Comment by erlend-aasland** [comment]
Thanks for the review!

**Pull request merged**